src/main/java/ch/njol/skript/sections/SecCatchErrors.java

```diff
@@ -0,0 +1,74 @@
+package ch.njol.skript.sections;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.config.SectionNode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Example;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.ExprCaughtErrors;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.Section;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.Trigger;
+import ch.njol.skript.lang.TriggerItem;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.log.runtime.RuntimeError;
+import org.skriptlang.skript.log.runtime.RuntimeErrorCatcher;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Name("Catch Runtime Errors")
+@Description("Catch any runtime errors produced by code within the section. This is an in progress feature")
+@Example("""
+	catch runtime errors:
+		set worldborder center of {_border} to location(0, 0, NaN value)
+	if last caught runtime errors contains "Your location can't have a NaN value as one of its components":
+		set worldborder center of {_border} to location(0, 0, 0)
+	""")
+@Since("INSERT VERSION")
+public class SecCatchErrors extends Section {
+
+	static {
+		Skript.registerSection(SecCatchErrors.class, "catch [run[ ]time] error[s]");
+	}
+
+	private Trigger trigger;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult, SectionNode sectionNode, List<TriggerItem> triggerItems) {
+		if (sectionNode.isEmpty()) {
+			Skript.error("A catch errors section must contain code.");
+			return false;
+		}
+
+		AtomicBoolean delayed = new AtomicBoolean(false);
+		Runnable afterLoading = () -> delayed.set(!getParser().getHasDelayBefore().isFalse());
+		trigger = loadCode(sectionNode, "runtime", afterLoading, Event.class);
```

Comment:
Given that the event context is not changing, you should not load this code into a trigger. You can simply load it as normal and check the updated delay value in ParserInstance. As for `walk`, to execute the section, you can do something like:
```java
last.setNext(null);
TriggerItem.walk(first, event);
```
`first` and `last` are available from TriggerSection and represent the first and last elements of the section, respectively.